### ISSUE TITLE
Doxygen XML-Output for C lang structs with sub-structs issue

### ISSUE THREAD

**Issue by Tom4096** [doxygen/doxygen#6652]
Structs with sub-structs are processed by doxygen as expected for output format HTML.

But for XML output format they are processed ambiguous as the following example shows.

**test.h**
```
typedef struct {
  struct {
    int   joe;
    short jack;
  } bar;
  struct {
    int   joe;
    short jack;
    long  bill;
  } saloon;
} a;

typedef struct {
  struct {
    int   joe;
    short jack;
  } bar;
  struct {
    int   joe;
    long  bill;
  } saloon;
} b;
```

**Doxyfile**
```
PROJECT_NAME           = Test
OPTIMIZE_OUTPUT_FOR_C  = YES
TYPEDEF_HIDES_STRUCT   = NO
EXTRACT_ALL            = YES
EXTRACT_LOCAL_CLASSES  = YES
FILE_PATTERNS          = *.h 
SOURCE_BROWSER         = NO
GENERATE_HTML          = YES
HTML_OUTPUT            = html
GENERATE_HTMLHELP      = NO
SERVER_BASED_SEARCH    = NO
GENERATE_LATEX         = NO
GENERATE_XML           = YES
XML_OUTPUT             = xml
```
Proceeding Doxygen will result result in two identical files (except struct name and generated ids) **structa.xml** and **structb.xml**.

This reported is based on [StackOver-Flow question](https://stackoverflow.com/questions/53536794/). So there might be further remarks to this issue reported.

**Comment by albert-github** [comment]
The problem here has, as far as I can see, to do with the anonymous structs. I think it is also to #5227 (Issues with template nullptr compatibility class (Origin: bugzilla #703262)).
I tested it with a bit an enhanced example (comments so we can better see what is happening) and with the current version (1.9.2) and the current master (1.9.3 (755cff695ec9f9d4f80725b9ed4081646cb164f5)) I think the example is also in HTML not correct as here I see:

![image](https://user-images.githubusercontent.com/5223533/145674536-613d8796-8dfd-483a-8bb2-a54ee29027d7.png)

In other words joe from the bar has been merged with joe the saloon. Also behind the ![image](https://user-images.githubusercontent.com/5223533/145674575-0edfcfbc-1a51-4e5a-a870-74222bb53080.png) of the "typedef" there is no name.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/7696881/example.tar.gz)

**Comment by albert-github** [comment]
@doxygen I ran with `doxygen -d variables` and saw:
```
VARIABLE_SEC:
  type='int' name='joe' args='' bodyLine=8 mGrpId=-1 relates=''
  functionPtr? no
  class variable:
    'int' 'a'::'joe' '' prot=0 ann=1 init=''
  class variable:
    'int' 'a::@0'::'joe' '' prot=0 ann=0 init=''
```

- is there a specific reason to have 2 variables for the variable `a.bar.joe`?
  - `a::joe` with `ann=1`
  - `a::@0::joe` with `ann=0`

  for `a.soloon.joe` we see:
  - `a::joe` with `ann=1`
  - `a::@1::joe` with `ann=0`

  hence the match from the 2 `a::joe`
